File: split/eqdefine.pas
procedure eqdefine(p: halfword; t: quarterword; e: halfword);
begin
  if eqtb[p].hh.b1 = curlevel then
    eqdestroy(eqtb[p])
  else if curlevel > 1 then
    eqsave(p, eqtb[p].hh.b1);
  eqtb[p].hh.b1 := curlevel;
  eqtb[p].hh.b0 := t;
  eqtb[p].hh.rh := e;
end;

File: cpp/eqdefine.cpp
#include "eqdefine.h"
#include "eqdestroy.h"
#include "eqsave.h"

void eqdefine(halfword p, quarterword t, halfword e)
{
	if (eq_level(p) == curlevel)
		eqdestroy(eqtb[p]);
	else 
		if (curlevel > 1)
			eqsave(p, eq_level(p));
	eq_level(p) = curlevel;
	eq_type(p) = t;
	equiv(p) = e;
}
